CI: separate fastlane and build code...WIP

0xbrayo opened the pull request:
<!-- ELLIPSIS_HIDDEN -->


| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit d54aed05255977357b1f79f4142d6c39bdedca4d  | 
|--------|--------|

### Summary:
Separated Fastlane and build code in CI, updated image conversion command, and added shell syntax highlighting in documentation.

**Key points**:
- Added `.github/workflows/build-only.yml` to separate Fastlane and build code in CI.
- Updated `Makefile` to use `magick` instead of `convert` for image conversion.
- Enabled `ExternalStorageConfigurationManager` in `.idea/misc.xml`.
- Added `sh` syntax highlighting to code block in `README.md`.


----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)



<!-- ELLIPSIS_HIDDEN -->

0xbrayo committed:
docs: update deprecated magick command

0xbrayo committed:
chore: add 'sh' to README.md

0xbrayo commented:
WIP. Separating the fastlane code from the build code

brayo-pip commented on .github/workflows/build-only.yml:
```diff
@@ -0,0 +1,234 @@
+      env:
+        KEY_ANDROID_JKS: ${{ secrets.KEY_ANDROID_JKS }}
+      run: |
+        printf "$KEY_ANDROID_JKS" > android.jks.key
+        cat android.jks.age | age -d -i android.jks.key -o android.jks
+        rm android.jks.key
```
What does this do? I don't understand where `android.jks.age` is initially created.

0xbrayo commented:
@ellipsis-dev review

0xbrayo committed:
chore: replace the default android.jks.age

0xbrayo committed:
refactor: refactored to improve accuracy of the timeline

konrad-krol-incubly committed:
feat: implemented generic web watcher with support of multiple web browsers

konrad-krol-incubly committed:
build(deps): updated project dependencies, jvm target to 11, gradle to 8.14

0xbrayo committed:
feat:use device name as hostname

0xbrayo committed:
chore: gradle version bump and misc

0xbrayo committed:
feat(android): add SyncInterface.kt with JNI bindings for aw-sync

- Created SyncInterface class to interact with native aw-sync library
- Implemented JNI bindings for sync operations:
  * syncPullAll - pull from all hosts
  * syncPush - push local data
  * syncBoth - full bidirectional sync
  * getSyncDir - get sync directory path
- Async wrappers with callbacks for all sync operations
- Automatic sync directory setup in external storage
- Proper error handling and logging

Phase 4 implementation: Android app integration

0xbrayo committed:
feat(android): implement automatic background sync

- Modified SyncInterface to use Downloads/ActivityWatch/ directory
  * Static path for easy access by sync apps (Syncthing, etc.)
  * Falls back gracefully if storage not available

- Created SyncScheduler for automatic periodic syncing
  * First sync runs 1 minute after app starts
  * Subsequent syncs every 15 minutes while app is open
  * Runs syncBoth (pull + push) for full bidirectional sync
  * Logs all sync operations and results

- Integrated into MainActivity lifecycle
  * Starts scheduler in onCreate after server initialization
  * Stops scheduler in onDestroy to clean up resources

No UI needed - syncs automatically in background.
Users can access sync directory at: /sdcard/Download/ActivityWatch/

Phase 4 complete: Automatic background sync implementation

0xbrayo committed:
ci: create build-only action for testing in forks

0xbrayo committed:
feat: always active in the background

0xbrayo committed:
chore: minimum api target bump

0xbrayo committed:
fix: fix launch on boot not working

0xbrayo committed:
feat: experimental parsing of events in background

0xbrayo committed:
build: update makefile to package aw-sync.so

0xbrayo committed:
fix: fix single-instance errors on android

0xbrayo committed:
fix: fix doNotStrip property not working

0xbrayo committed:
Update mobile/src/main/java/net/activitywatch/android/watcher/WebWatcher.kt

Co-authored-by: ellipsis-dev[bot] <65095814+ellipsis-dev[bot]@users.noreply.github.com>

0xbrayo committed:
style: fix linting warrnings

0xbrayo committed:
temp: switch to custom aw-webui

0xbrayo committed:
wip

0xbrayo committed:
checkpoint

0xbrayo committed:
checkpoint 2

0xbrayo committed:
checkpoint widget

0xbrayo committed:
checkpoint widget 2

0xbrayo committed:
checkpoint widget stacked

0xbrayo committed:
4x1 checkpoint

0xbrayo committed:
animation + reparsing

0xbrayo committed:
savepoint

0xbrayo committed:
fix: use upstream aw-server-rust commit (dc70318) to avoid broken aw-webui reference

0xbrayo committed:
fix: restore aw-server-rust to babea6c (working version)

0xbrayo committed:
TEMP: ci fix, adds release

0xbrayo committed:
update aw-server-rust to latest

0xbrayo committed:
temp: use fix-webui-ref-rebase branch from fork to test CI fix

This temporarily uses the fix-webui-ref-rebase branch from the fork which
updates aw-webui to a valid commit (64ed7a7) that exists in the remote.
This should fix CI failures related to invalid aw-webui submodule reference.

0xbrayo committed:
feat: media watcher
